v2/cli/init_command.go

```diff
@@ -0,0 +1,248 @@
+		},
+	},
+	Action: initialize,
+}
+
+// Handle init CLI flags
+func initialize(c *cli.Context) error {
+
+	// Initialize fields to fill in the template yaml files
+	fields := InitFields{
+		ID:       c.String(flagID),
+		Language: strings.ToLower(c.String(flagLanguage)),
+		Git:      c.String(flagGit),
+	}
+
+	err := idMatch(fields.ID)
+	if err != nil {
+		return err
+	}
+
+	// Check if input language is valid
+	fields.Language, err = normalizeLanguage(fields.Language)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade directory exists
+	if dirExists, dirErr := file.Exists(".brigade"); dirErr != nil {
+		return dirErr
+	} else if !dirExists {
+		fmt.Println("Creating .brigade directory...")
+		err = os.Mkdir(".brigade", 0755)
+		if err != nil {
+			return err
+		}
+	}
+
+	buf, err := execTemplate(newProjectFileBytes, fields)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade/project.yaml path exists.
+	// Confirm overwrite if so.
+	if err = checkFileOverwrite(
+		path.Join(".brigade", "project.yaml"),
+		"project.yaml",
+		buf,
+	); err != nil {
+		return err
+	}
+
+	// If git repo is provided, create a separate js/ts file
+	if fields.Git != "" {
+		gitBuf, gitErr :=
+			execTemplate(newGitBrigadeFileBytes, fields)
+		if gitErr != nil {
+			return gitErr
+		}
+
+		gitFileName := fmt.Sprintf("brigade.%s", fields.Language)
+
+		if err = checkFileOverwrite(
+			path.Join(".brigade", gitFileName),
+			gitFileName,
+			gitBuf,
+		); err != nil {
+			return err
+		}
+	}
+
+	// TODO: Right now the notes doesn't have template fields, add later!
+	notesBuf, err :=
+		execTemplate(newNotesFileBytes, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a NOTES.txt file
+	if err = checkFileOverwrite(
+		path.Join(".brigade", "NOTES.txt"),
+		"NOTES.txt",
+		notesBuf,
+	); err != nil {
+		return err
+	}
+
+	secretsBuf, err :=
+		execTemplate(newSecretsFileBytes, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a secrets.yaml file
+	if err = checkFileOverwrite(
+		path.Join(".brigade", "secrets.yaml"),
+		"secrets.yaml",
+		secretsBuf,
+	); err != nil {
+		return err
+	}
+
+	err = addToGitignore(".brigade/secrets.yaml")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Detect if .gitignore exists. If it does, add .brigade/secrets.yaml to it
+// If not, create it and add .brigade/secrets.yaml to it
+func addToGitignore(filepath string) error {
+	if fileExists, err := file.Exists("./.gitignore"); err != nil {
+		return err
+	} else if fileExists {
+		f, err :=
+			os.OpenFile(".gitignore", os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+		if err != nil {
+			return err
+		}
+
+		defer f.Close()
+
+		b, err := ioutil.ReadFile(".gitignore")
+		if err != nil {
+			return err
+		}
+		s := string(b)
+
+		// Only add to gitignore if it's not in the gitignore already
+		if !strings.Contains(s, filepath) {
+			if _, err = f.WriteString("\n" + filepath); err != nil {
+				return err
+			}
+		}
```

Comment:
I think there are probably a lot of weird cases here that weren't anticipated.

If this function were used to ignore "foo" and "foobar" were already in the list, we'd end up _not_ ignoring "foo". Another weird example is that if "foo/" were in the list already, "foo" wouldn't get added and those two things aren't equivalent ("foo/" matches everything under a _directory_ named "foo/").

I think an acceptable route is to unconditionally add the indicated entry and have the comment at the top of the function explicitly note that it's unconditional.

Reply:
Which simplifies the logic for this whole function to "create .gitignore if it doesn't exist then, either way, open that file in append mode to add to it."